module/os/linux/zfs/zvol_os.c

```diff
@@ -83,6 +117,108 @@ zv_request_task_free(zv_request_task_t *task)
 	kmem_free(task, sizeof (*task));
 }
 
+#ifdef HAVE_BLK_MQ
+/*
+ * This is our blk-mq workqueue callback function.  It's here that
+ * we process all the BIOs in a request.
+ */
+static void zvol_mq_work_func(struct work_struct *work)
+{
+	zv_work_t *zv_work;
+	struct bio *bio = NULL;
+	struct request *rq;
+	zvol_state_t *zv;
+	blk_status_t res = BLK_STS_OK;
+
+	zv_work = container_of(work, zv_work_t, work);
+	rq = zv_work->rq;
+	zv = rq->q->queuedata;
+
+	/* Tell the kernel that we are starting to process this request */
+	blk_mq_start_request(rq);
+
+	if (blk_rq_is_passthrough(rq)) {
+		/* Skip non filesystem request */
+		blk_mq_end_request(rq, BLK_STS_IOERR);
+		goto out;
```

Comment:
Personally the `goto` here feels a bit overkill to me.  I think it'd be clearer to just `kmem_cache_free` and `return` here.